tensorboard/webapp/feature_flag/views/feature_flag_page_component_test.ts

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {CommonModule} from '@angular/common';
+import {Component, NO_ERRORS_SCHEMA} from '@angular/core';
+import {TestBed} from '@angular/core/testing';
+import {FeatureFlags} from '../types';
+import {FeatureFlagPageComponent} from './feature_flag_page_component';
+import {FeatureFlagOverrideStatus, FeatureFlagStatus} from './types';
+
+describe('feature_flag_page_component', () => {
+  async function createComponent(
+    featureFlagStatuses: FeatureFlagStatus<keyof FeatureFlags>[]
+  ) {
+    @Component({
+      selector: 'testable-component',
+      template: `<feature-flag-page-component
+        [featureFlagStatuses]="featureFlagStatuses"
+      >
+      </feature-flag-page-component>`,
+    })
+    class TestableComponent {
+      get featureFlagStatuses(): FeatureFlagStatus<keyof FeatureFlags>[] {
+        return featureFlagStatuses;
+      }
+    }
+
+    await TestBed.configureTestingModule({
+      declarations: [TestableComponent, FeatureFlagPageComponent],
+      imports: [CommonModule],
+      schemas: [NO_ERRORS_SCHEMA],
+    }).compileComponents();
+
+    const fixture = TestBed.createComponent(TestableComponent);
+    fixture.detectChanges();
+    return fixture.nativeElement.querySelector('feature-flag-page-component');
+  }
+
+  it('creates rows for each feature flag', async () => {
```

Comment:
Normally we prefer to have a single test file for each container/component combination. We think it strikes a good balance between writing isolated tests but also ensuring a particular container/component combination works as expected when integrated together. The fact that they are split into a container/component is (mostly) an implementation detail that we hide from the tests.

In this case we would have a feature_flag_page_test instead of both feature_flag_page_component_test and feature_flag_page_container_test.

I mention it here specifically because for these tests where you introduce TestableComponent, you could instead test by constructing a FeatureFlagPageContainer and specifying appropriate values for your ngrx selectors and then checking that the UI renders as expected.